OpenRA.Mods.Common/Traits/World/MPStartLocations.cs

```diff
@@ -89,7 +89,10 @@ static CPos ChooseSpawnPoint(World world, List<CPos> available, List<CPos> taken
 			if (available.Count == 0)
 				throw new InvalidOperationException("No free spawnpoint.");
 
-			var n = taken.Count == 0
+			var spawnpointoptions = world.WorldActor.TraitOrDefault<SpawnPointOption>();
+			var fullyRandomSpawnPos = spawnpointoptions != null ? spawnpointoptions.Enabled : false;
```

Comment:
`var fullyRandomSpawnPos = spawnpointoptions != null && spawnpointoptions.Enabled;`

You also want to have camelCase `spawnPointOptions`.